stubs/greenlet/greenlet/_greenlet.pyi

```diff
@@ -30,10 +34,11 @@ class greenlet:
     def gr_context(self, value: Context | None) -> None: ...
     @property
     def gr_frame(self) -> FrameType | None: ...
-    @property
-    def parent(self) -> greenlet | None: ...
-    @parent.setter
-    def parent(self, value: greenlet | None) -> None: ...
+    # the parent attribute is a bit special, since it can't be set to `None` manually, but
+    # it can be `None` for the master greenlet which will always be around, regardless of
+    # how many greenlets have been spawned explicitly. Since there can only be one such
+    # greenlet per thread, there is no way to create another one manually.
+    parent = _ParentDescriptor
```

Comment:
You want it to be an instance of `_ParentDescriptor` rather than the `_ParentDescriptor` class itself, right?

```suggestion
    parent: _ParentDescriptor
```